// Copyright 2018 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package zucchini.fuzzers;

// NEXT_TAG = 4
message FilePair {
  // File to generate patch from or apply patch to.
  required bytes old_file = 1;
  // New file to generate patch or the patch to apply.
  required bytes new_or_patch_file = 2;
  // Imposed matches to apply to the equivalence matches.
  // Should be of the format:
  //   "#+#=#+#,#+#=#+#,..." (e.g., "1+2=3+4", "1+2=3+4,5+6=7+8"),
  // where "#+#=#+#" encodes a match as 4 unsigned integers:
  //   [offset in "old", size in "old", offset in "new", size in "new"].
  optional string imposed_matches = 3;
}
